src/core/SpriteAnimator.tsx

```diff
@@ -102,23 +155,22 @@ export const SpriteAnimator: React.FC<SpriteAnimatorProps> = /* @__PURE__ */ Rea
     },
     fref
   ) => {
-    const ref = React.useRef<any>()
-    const spriteData = React.useRef<any>(null)
-    const matRef = React.useRef<any>()
-    const spriteRef = React.useRef<any>()
+    const ref = React.useRef(new THREE.Group())
+    const spriteData = React.useRef<SpriteData | null>(null)
+    const matRef = React.useRef<THREE.MeshBasicMaterial | null>(null)
+    const spriteRef = React.useRef<THREE.Mesh | THREE.InstancedMesh>(null)
     const timerOffset = React.useRef(window.performance.now())
-    const currentFrame = React.useRef<number>(startFrame || 0)
-    const currentFrameName = React.useRef<string>(frameName || '')
-    const fpsInterval = (fps ?? 30) > 0 ? 1000 / (fps || 30) : 0
-    const [spriteTexture, setSpriteTexture] = React.useState<THREE.Texture>(new THREE.Texture())
-    const totalFrames = React.useRef<number>(0)
-    const [aspect, setAspect] = React.useState<number[] | null>([1, 1, 1])
+    const currentFrame = React.useRef(startFrame)
+    const currentFrameName = React.useRef(frameName)
+    const fpsInterval = fps > 0 ? 1000 / fps : 0
+    const [spriteTexture, setSpriteTexture] = React.useState(new THREE.Texture())
+    const totalFrames = React.useRef(0)
+    const [aspect, setAspect] = React.useState(new THREE.Vector3(1, 1, 1))
     const flipOffset = flipX ? -1 : 1
-    const [displayAsSprite, setDisplayAsSprite] = React.useState(asSprite ?? true)
+    const [displayAsSprite, setDisplayAsSprite] = React.useState(asSprite)
```

Comment:
you can remove that state, which is just syncing its value with `asSprite`: instead you can just use `asSprite` value direclty